### PR TITLE
[RLlib]: Fix multibinary space serialization

### LINKED ISSUE #34156: [RLlib] MultiBinary spaces not deserializable
### What happened + What you expected to happen

I am using a MultiBinary space as part of the observations space in a custom environment. The serialization of the algorithm fails when checkpointing due to a bug in the way MultiBinary spaces are translated to dict and viceversa.

The bug is due to the use of a `dtype` argument that is not accepted by MultiBinary's `__init__`

### Versions / Dependencies

ray == 2.3.1
gymnasium == 0.26.3

### Reproduction script

```python
import gymnasium as gym
from ray.rllib.utils.serialization import space_from_dict,space_to_dict
s = gym.spaces.MultiBinary((2,3))
d = space_to_dict(s)
# d= {'space': {'space': 'multi-binary', 'n': (2, 3), 'dtype': '|i1'}}
s_ = space_from_dict(d)
```

This will raise the following exception:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[21], line 1
----> 1 s_ = space_from_dict(d)

File ~/.../lib/python3.8/site-packages/ray/rllib/utils/serialization.py:281, in space_from_dict(d)
    279 @DeveloperAPI
    280 def space_from_dict(d: Dict) -> gym.spaces.Space:
--> 281     space = gym_space_from_dict(d["space"])
    282     if "original_space" in d:
    283         space.original_space = gym_space_from_dict(d["original_space"])

File ~/.../lib/python3.8/site-packages/ray/rllib/utils/serialization.py:276, in gym_space_from_dict(d)
    273 if space_type not in space_map:
    274     raise ValueError("Unknown space type for de-serialization, ", space_type)
--> 276 return space_map[space_type](d)

File ~/.../lib/python3.8/site-packages/ray/rllib/utils/serialization.py:226, in gym_space_from_dict.<locals>._multi_binary(d)
    225 def _multi_binary(d: Dict) -> gym.spaces.MultiBinary:
--> 226     return gym.spaces.MultiBinary(**__common(d))

TypeError: __init__() got an unexpected keyword argument 'dtype'
```

### Issue Severity

High: It blocks me from completing my task.

### PR DESCRIPTION
**Pull request by Rohan138**
## Related issue number

Closes #34156 

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [X] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/utils/serialization.py [hunk 1]
```diff
@@ -113,8 +113,7 @@ def _discrete(sp: gym.spaces.Discrete) -> Dict:
     def _multi_binary(sp: gym.spaces.MultiBinary) -> Dict:
         return {
             "space": "multi-binary",
-            "n": int(sp.n),
-            "dtype": sp.dtype.str,
+            "n": sp.n,
```
Comment (avnishn) [inline rllib/utils/serialization.py:116]: is sp.n an integer already, or is it an np.int. 

if it is an np.int it can't be serialized.

**Review by avnishn — COMMENTED** [review]
Left a question

File: rllib/utils/serialization.py (hunk 1 above)
Comment (Rohan138) [reply rllib/utils/serialization.py:116]: integer or tuple of ints, actually

**Comment by kouroshHakha** [comment]
Tests are passing ... Merging ...

**Review by kouroshHakha — APPROVED** [review]

**Pull request merged**